### PR TITLE
feat: Use widget features during registration

### PR DESCRIPTION
**Pull request by riodeuno**
## Description
- [ ] Use widget features to enhance property pane configurations
- [ ] Use widget features to add default values to widget when they're created

## Type of change
- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?
- Manual
- Jest PENDING

### CODE REVIEW

File: app/client/src/utils/WidgetRegisterHelpers.tsx [hunk 1]
```diff
@@ -50,9 +54,17 @@ export const registerWidget = (Widget: any, config: WidgetConfiguration) => {
...
+        {},
+        WidgetFeatureProps[registeredFeature as RegisteredWidgetFeatures],
+        WidgetFeaturePropertyEnhancements[
+          registeredFeature as RegisteredWidgetFeatures
+        ](config),
+      );
+    });
   }
   const _config = {
     ...features,
```
Comment (riodeuno) [inline app/client/src/utils/WidgetRegisterHelpers.tsx:70]: Move this spread below ...config.defaults, as we want the features to override the defaults.

File: app/client/src/utils/WidgetRegisterHelpers.tsx [hunk 2]
```diff
@@ -50,13 +54,21 @@ export const registerWidget = (Widget: any, config: WidgetConfiguration) => {
 };
 
 export const configureWidget = (config: WidgetConfiguration) => {
-  let features = {};
-  if (config.features && config.features.dynamicHeight) {
-    features = Object.assign({}, WidgetFeatureProps.DYNAMIC_HEIGHT);
+  let features: Record<string, unknown> = {};
+  if (config.features) {
+    Object.keys(config.features).forEach((registeredFeature: string) => {
```
Comment (marks0351) [inline app/client/src/utils/WidgetRegisterHelpers.tsx:59]: suggestion: you cud use reduce instead of forEach.

**Review by marks0351 — APPROVED** [review]

File: app/client/src/utils/WidgetRegisterHelpers.tsx (hunk 2 above)
Comment (riodeuno) [reply app/client/src/utils/WidgetRegisterHelpers.tsx:59]: @marks0351 could you elaborate on the advantages here? IMO, forEach is easier to grok.

Comment (marks0351) [reply app/client/src/utils/WidgetRegisterHelpers.tsx:59]: it's just that reduce is the actual function that does the exact same thing that is being done with the variable initiation and forEach. for me reduce and forEach both are readable. also there are no special advantages coz of this.

**Review by rahulramesha — APPROVED** [review]

**Pull request merged**